packages/app/components/stack-modal/StackModal.tsx

```diff
@@ -32,100 +31,226 @@ import {
   totalFundsUsed,
 } from "@/models/stack-order";
 import { formatTokenValue } from "@/utils/token";
+import { getDCAOrderContract } from "@stackly/sdk";
+import { useEthersSigner } from "@/utils/ethers";
+import { convertedAmount } from "@/utils/numbers";
+import {
+  StackedTokenLogoPair,
+  DialogConfirmTransactionLoading,
+  FromToStackTokenPair,
+} from "@/components";
+import { StackProgress } from "@/components/stack-modal/StackProgress";
+import { StackOrdersTable } from "@/components/stack-modal/StackOrdersTable";
 
 interface StackModalProps extends ModalBaseProps {
   stackOrder: StackOrder;
+  refetchStacks: () => void;
 }
 
+export const txEplorerLink = (tx: string) => `https://gnosisscan.io/tx/${tx}`;
+
 export const transactionExplorerLink = (address: string) =>
   `https://gnosisscan.io/address/${address}#tokentxns`;
 
+interface CancellationTransaction {
+  hash: string;
+}
+
 export const StackModal = ({
   stackOrder,
   isOpen,
+  refetchStacks,
   closeAction,
 }: StackModalProps) => {
+  const signer = useEthersSigner();
+
+  const [isOpenCancellationDialog, setIsOpenCancellationDialog] =
+    useState(false);
+  const [isCancellationProcessing, setIsCancellationProcessing] =
+    useState(false);
+  const [isCancellationSuccess, setIsCancellationSuccess] = useState(false);
+  const [cancellationTx, setCancellationTx] =
+    useState<CancellationTransaction>();
+
   const orderSlots = stackOrder.orderSlots;
   const firstSlot = orderSlots[0];
   const lastSlot = orderSlots[orderSlots.length - 1];
   const nextSlot = orderSlots[totalOrdersDone(stackOrder)];
-  const stackIsComplete = totalOrdersDone(stackOrder) === orderSlots.length;
+
+  const remainingFunds =
+    convertedAmount(stackOrder.amount, stackOrder.buyToken.decimals) -
+    totalFundsUsed(stackOrder);
+
+  const remainingFundsDescription =
+    remainingFunds > 0 &&
+    `remaining funds, ${remainingFunds} ${stackOrder.sellToken.symbol}`;
+
+  const stackIsComplete =
+    totalOrdersDone(stackOrder) === orderSlots.length && remainingFunds === 0;
+
+  const cancelStack = async () => {
+    if (!signer) return;
+
+    try {
+      setIsCancellationProcessing(true);
+      const tx = await getDCAOrderContract(stackOrder.id, signer).cancel();
+      setCancellationTx(tx);
+      await tx.wait().then(() => {
+        setIsCancellationProcessing(false);
+        setIsCancellationSuccess(true);
+      });
+    } catch (e) {
+      setIsCancellationProcessing(false);
+      console.error("error", e);
```

Comment:
I think we can have a cleaner message. Or no  "error" message at all, as it's already a console error message.

Reply:
I would put "cancel stack error"